### DISCUSSION TITLE
How can I find out about a reconnection or change in connect id?

### DISCUSSION THREAD

**Discussion by MrNapcae** [pocketbase/pocketbase#5600] [Q&A]
Is there a way to find out about the current reconnect or change of clientId? 
`pb.realtime.subscribe('PB_CONNECT')` does not provide information about reconnection and clientId change. 
How can one find out about the change of clientId in case of a long-term connection or its interruption?

**Comment by ganigeorgiev** [comment]
It is not clear what you are asking. The `PB_CONNECT` event is fired on every connect/reconnect and a new random `clientId` is generated on each connection. The `clientId` value shouldn't matter for the end user.

Please elaborate a little more on your use case to understand what you are trying to do.

**Reply by MrNapcae** [reply]
Thanks for the quick response, if the application (react-native) is inactive for a long time, the token becomes irrelevant and the subscriptions that were originally returned with the error No client associated with connection ID. I try to update subscriptions if the connection ID changes. pb.realtime.clientId causes endless updates if you subscribe to it...

**Reply by ganigeorgiev** [reply]
I don't understand what do you mean, especially _"I try to update subscriptions if the connection ID changes"_.

The JS SDK should take care to reconnect automatically. Again, the `clientId` is random for each new connection. Once the SSE connection is interrupted, the associated `clientId` is discarded.

I'll stop engaging to avoid further confusion, but if you need someone to assist you try to explain more clearly what actually is the issue you are experiencing and ideally provide a minimal code sample with the problematic part (_see also https://xyproblem.info/_).
